src/sentry/api/endpoints/project_repo_path_parsing.py

```diff
@@ -56,7 +56,11 @@ def org_id(self):
     def validate_source_url(self, source_url: str):
         # first check to see if we are even looking at the same file
         stack_path = self.initial_data["stack_path"]
-        stack_file = stack_path.split("/")[-1]
+
+        if "\\" in stack_path:
+            stack_file = stack_path.split("\\")[-1]
+        else:
+            stack_file = stack_path.split("/")[-1]
```

Comment:
```suggestion
            stack_file = pathlib.PureWindowsPath(stack_path).name
```

Reply:
Much cleaner solution, thanks @vartec!

This will convert the stack root to use forward slashes though - should this be the expected behavior or do we want the slashes in the stack root to match what the user uses?
![image](https://github.com/getsentry/sentry/assets/55160142/2cdf9c94-b6a0-455a-8384-6b585c0a8463)

Reply:
nvm i think i misunderstood the change you were suggesting - this shouldn't affect the testing results